pandas/core/strings.py

```diff
@@ -109,9 +114,46 @@ def cat_safe(list_of_columns: List, sep: str):
 
 def _na_map(f, arr, na_result=np.nan, dtype=object):
     # should really _check_ for NA
+    if is_extension_array_dtype(arr.dtype):
+        return _map_ea(f, arr, na_value=na_result, dtype=dtype)
     return _map(f, arr, na_mask=True, na_value=na_result, dtype=dtype)
 
 
+def _map_ea(f, arr, na_value, dtype):
```

Comment:
nitpick: `f` -> `func`, annotattion for `dtype` (in particular, is it just np.dtype or can it be EADtype?)

Since this is internal, I'd advocate typing `arr` as having already had extract_array called on it